lib/datadog/core/remote/configuration/target.rb

```diff
@@ -9,34 +10,43 @@ module Remote
       class Configuration
         # TargetMap stores information regarding Configuration::Path and Configuration::Target
         class TargetMap < Hash
+          class ParseError < StandardError; end
+
           class << self
             def parse(hash)
-              opaque_backend_state = hash['signed']['custom']['opaque_backend_state']
-              version = hash['signed']['version']
+              signed = hash['signed']
+
+              opaque_backend_state = signed['custom']['opaque_backend_state']
+              version = signed['version']
+              expires = signed['expires']
 
               map = new
 
               map.instance_eval do
                 @opaque_backend_state = opaque_backend_state
                 @version = version
+                @expires = DateTime.iso8601(expires) if expires
```

Comment:
Note that according to the [Ruby documentation](https://rubyapi.org/3.2/o/datetime#method-c-iso8601):

> [DateTime](https://rubyapi.org/3.2/o/datetime) class is considered deprecated. Use [Time](https://rubyapi.org/3.2/o/time) class.

:sweat_smile: